app/api/crud.py

```diff
@@ -135,30 +135,36 @@ async def post_subjects(
         Response of the POST request.
 
     """
-    nodes = util.validate_queried_nodes(query.get("nodes"))
-    node_urls = [node["node_url"] for node in nodes]
+    node_datasets_filters = util.validate_queried_nodes(query.get("nodes"))
+    node_urls = [node["node_url"] for node in node_datasets_filters]
 
     query.pop("nodes", None)
 
     tasks = []
     # NOTE: Nodes in a single request can only be ALL dicts
-    for node in nodes:
-        node_request_url = node["node_url"] + "subjects"
-        query["dataset_uuids"] = node.get("dataset_uuids")
+    for node_datasets_filter in node_datasets_filters:
+        node_request_url = node_datasets_filter["node_url"] + "subjects"
+        # Ensure each task gets its own copy of the base query.
+        # Otherwise, mutating the original dict would cause all tasks to share the same final dataset_uuids value
+        # since 'query' is passed by reference.
+        node_query = query.copy()
```

Comment:
While this fixes the bug, the copy is shallow here, so the problem could come back for nested attributes. I'd say the deeper issue here is the mutation of the query within the request function - post_datasets is similar, just hasn't encountered this issue yet.